helper/resource/testing_new.go

```diff
@@ -436,18 +436,6 @@ func runNewTest(ctx context.Context, t testing.T, c TestCase, helper *plugintest
 				t.Fatalf("Error generating merged configuration: %s", err)
 			}
 
-			confRequest := teststep.PrepareConfigurationRequest{
-				Directory: step.ConfigDirectory,
-				File:      step.ConfigFile,
-				Raw:       mergedConfig,
-				TestStepConfigRequest: config.TestStepConfigRequest{
-					StepNumber: stepIndex + 1,
-					TestName:   t.Name(),
-				},
-			}.Exec()
-
-			appliedCfg = teststep.Configuration(confRequest)
-
```

Comment:
It looks like this was only ever needed and done for the import tests, so I've opted to move this into `testing_new_import_state.go` and pass the `appliedConfig` through as a string so it can be easily appended/modified based on the import mode/kind we're using in the test.